### PR TITLE
DPC-21: Add ability to remove attribution relationship

### PR DESCRIPTION
**Pull request by nickrobison-usds**
This PR resolves DPC-21 and adds the ability to remove attribution relationships between a given patient and provider.

Actually adding a specific relationship will be handled in DPC-179.

This should also close DPC-90, the Attribution Engine now requires that all method parameters be FHIR resources.

### CODE REVIEW

File: dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/AggregationEngine.java [hunk 1]
```diff
@@ -54,7 +55,7 @@ public void run() {
                 final JobModel model = (JobModel) workPair.get().getRight();
                 final UUID jobID = workPair.get().getLeft();
                 logger.debug("Has job {}. Working.", jobID);
-                Set<String> attributedBeneficiaries = model.getBeneficiaries();
+                List<String> attributedBeneficiaries = model.getBeneficiaries();
```
Comment (isears-cms) [inline dpc-aggregation/src/main/java/gov/cms/dpc/aggregation/engine/AggregationEngine.java]: In keeping with the [DPC-126](https://jiraent.cms.gov/browse/DPC-126) terminology decision, would it be appropriate to refactor this name to `attributedPatients`?

File: dpc-attribution/src/test/java/gov/cms/dpc/attribution/AttributionResourceTest.java [hunk 2]
```diff
@@ -63,30 +64,28 @@ public void testBasicAttributionFunctionality() throws IOException {
...
+
+            final HttpGet notAttributed = new HttpGet("http://localhost:" + APPLICATION.getLocalPort() + "/v1/Group/0c527d2e-2e8a-4808-b11d-0fa06baf8254/19990000002901");
+            notAttributed.setHeader("Accept", FHIRMediaTypes.FHIR_JSON);
+
+            try (CloseableHttpResponse response = client.execute(notAttributed)) {
+                assertEquals(HttpStatus.NOT_ACCEPTABLE_406, response.getStatusLine().getStatusCode(), "Should be attributed");
+            }
 
 //            // Add them back
 //            final HttpPut httpCreate = new HttpPut("http://localhost:" + APPLICATION.getLocalPort() + "/v1/Group/0c527d2e-2e8a-4808-b11d-0fa06baf8254/19990000002901");
```
Comment (isears-cms) [inline dpc-attribution/src/test/java/gov/cms/dpc/attribution/AttributionResourceTest.java]: Does this commented-out code block need to stay in for another future ticket?

File: dpc-attribution/src/main/resources/application.conf [hunk 3]
```diff
@@ -4,6 +4,13 @@ dpc.attribution {
     url = "jdbc:h2:file:./h2/app_db;DB_CLOSE_ON_EXIT=FALSE"
   }
 
+  logging {
+    loggers {
+      "gov.cms.dpc" = DEBUG
+      "org.hibernate.SQL" = DEBUG
```
Comment (isears-cms) [inline dpc-attribution/src/main/resources/application.conf]: Do we need DEBUG-level logging for hibernate?

File: dpc-attribution/src/test/java/gov/cms/dpc/attribution/AttributionResourceTest.java (hunk 2 above)
Comment (nickrobison-usds) [reply dpc-attribution/src/test/java/gov/cms/dpc/attribution/AttributionResourceTest.java]: That was the plan, but I can remove it if we want.

File: dpc-attribution/src/main/resources/application.conf (hunk 3 above)
Comment (nickrobison-usds) [reply dpc-attribution/src/main/resources/application.conf]: Good catch. Nope!

File: dpc-attribution/src/test/java/gov/cms/dpc/attribution/AttributionResourceTest.java (hunk 2 above)
Comment (isears-cms) [reply dpc-attribution/src/test/java/gov/cms/dpc/attribution/AttributionResourceTest.java]: Nah, just making sure it wasn't left in by mistake

**Pull request merged**